src/functional_components/DriveMenu/DriveMenu.js

```diff
@@ -0,0 +1,145 @@
+import React, { useRef } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+import styles from './DriveMenu.module.scss';
+import { logout } from '../../actions/userActions';
+import {
+    closeDriveMenu,
+    copyItems,
+    pasteItems,
+    openConsentWindow,
+    openCreateFileWindow,
+    openCreateFolderWindow,
+    openRenameWindow,
+} from '../../actions/appActions';
+import useClickOutside from '../../hooks/useClickOutside';
+import Trash2 from '../../assets/svgIcons/Trash2';
+import Edit3 from '../../assets/svgIcons/Edit3';
+import Copy from '../../assets/svgIcons/Copy';
+import Clipboard from '../../assets/svgIcons/Clipboard';
+import FolderPlus from '../../assets/svgIcons/FolderPlus';
+import FilePlus from '../../assets/svgIcons/FilePlus';
+import Logout from '../../assets/svgIcons/LogOut';
+
+export const DriveMenu = ({
+    isDriveMenuVisible,
+    selectedItems,
+    renderOption,
+    closeDriveMenu,
+    clipboard,
+    pasteItems,
+    copyItems,
+    currentPath,
+    openConsentWindow,
+    openCreateFileWindow,
+    openCreateFolderWindow,
+    openRenameWindow,
+    logout,
+}) => {
+    const defaultRender = (option) => {
+        return (
+            <div
+                key={option.label}
+                onClick={
+                    !option.disabled
+                        ? () => {
+                              option.onClick();
+                              closeDriveMenu();
+                          }
+                        : undefined
+                }
+                className={classNames(styles.option, {
+                    [styles.disabled]: option.disabled,
+                    [styles.red]: !option.disabled && option.red,
+                })}
+            >
+                {option.icon}
+                <div className={styles.optionLabel}>{option.label}</div>
+            </div>
+        );
+    };
+
+    const options = [
+        {
+            label: 'Copy',
+            onClick: () => {
+                copyItems(selectedItems);
+            },
+            disabled: !selectedItems || selectedItems.length === 0,
+            icon: <Copy />,
+        },
+        {
+            label: 'Paste',
+            onClick: () => pasteItems(clipboard, currentPath),
+            disabled: !clipboard || clipboard.length === 0,
+            icon: <Clipboard />,
+        },
+        {
+            label: 'Rename',
+            onClick: () => openRenameWindow,
+            disabled: !selectedItems || selectedItems.length !== 1,
+            icon: <Edit3 />,
+        },
+        {
+            label: 'Create Folder',
+            onClick: () => openCreateFolderWindow,
```

Comment:
I Found a bug, as it is currently it doesen't open the create folder dialog.

it can be fixed with:
```suggestion
            onClick: openCreateFolderWindow,
```